tslearn/tests/test_serialize_models.py

```diff
@@ -0,0 +1,253 @@
+
+
+try:
+    os.makedirs(tmp_dir)
+except (FileExistsError, OSError):
+    pass
+
+
+def teardown_module():
+    clear_tmp()
+    os.removedirs(tmp_dir)
+
+
+def clear_tmp():
+    files = glob(os.path.join(tmp_dir, '*'))
+    for f in files:
+        os.remove(f)
+
+
+def test_hdftools():
+    dtypes = [numpy.int, numpy.int8, numpy.int16, numpy.int32, numpy.int64,
+              numpy.float, numpy.float32, numpy.float64]
+
+    d = {}
+
+    for dtype in dtypes:
+        name = numpy.dtype(dtype).name
+        d[name] = (numpy.random.rand(100, 100) * 10).astype(dtype)
+
+    fname = os.path.join(tmp_dir, 'hdf_test.hdf5')
+
+    hdftools.save_dict(d, filename=fname, group='data')
+
+    d2 = hdftools.load_dict(fname, 'data')
+
+    for k in d2.keys():
+        numpy.testing.assert_equal(d[k], d2[k])
+
+
+def _check_not_fitted(model):
+    # not serializable if not fitted
+    for fmt in all_formats:
+        with pytest.raises(NotFittedError):
+            getattr(model, "to_{}".format(fmt))(
+                os.path.join(
+                    tmp_dir, "{}.{}".format(model.__class__.__name__, fmt)
+                )
+            )
+
+
+def _check_params_predict(model, X):
+    # serialize to all all_formats
+    for fmt in all_formats:
+        getattr(model, "to_{}".format(fmt))(
+            os.path.join(
+                tmp_dir, "{}.{}".format(model.__class__.__name__, fmt)
+            )
+        )
+
+    # loaded models should have same model params
+    # and provide the same predictions
+    for fmt in all_formats:
+        sm = getattr(model, "from_{}".format(fmt))(
+            os.path.join(
+                tmp_dir, "{}.{}".format(model.__class__.__name__, fmt)
+            )
+        )
+
+        # make sure it's restored to the same class
+        assert isinstance(sm, model.__class__)
+
+        # test that saved model gives same predictions
+        numpy.testing.assert_equal(model.predict(X), sm.predict(X))
+
+        # check that the model-params are the same
+        model_params = model._get_model_params()
+        for p in model_params.keys():
+            numpy.testing.assert_equal(getattr(model, p), getattr(sm, p))
+
+        # check that hyper-params are the same
+        hyper_params = model.get_params()
+        for p in hyper_params .keys():
+            numpy.testing.assert_equal(getattr(model, p), getattr(sm, p))
+
+    clear_tmp()
+
+
+def test_serialize_global_alignment_kernel_kmeans():
+    seed = 0
+    numpy.random.seed(seed)
+    X_train, y_train, X_test, y_test = CachedDatasets().load_dataset("Trace")
+    # Keep first 3 classes
+    X_train = X_train[y_train < 4]
+    numpy.random.shuffle(X_train)
+    # Keep only 50 time series
+    X_train = TimeSeriesScalerMeanVariance().fit_transform(X_train[:50])
+    sz = X_train.shape[1]
+
+    gak_km = GlobalAlignmentKernelKMeans(n_clusters=3,
+                                         sigma=sigma_gak(X_train),
+                                         n_init=20,
```

Comment:
`n_init=1` is enough, we don't want to check that the results are good, we just want to check that there are consistent